blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -185,6 +175,36 @@ public Link getLink() {
         }
     }
 
+    protected String createCredentialId(@Nonnull String apiUrl) {
+        return ID;
+    }
+
+    protected @Nonnull String getCustomApiUri() {
+        StaplerRequest request = Stapler.getCurrentRequest();
+        Preconditions.checkNotNull(request, "Must be called in HTTP request context");
+        String apiUri = request.getParameter("apiUrl");
+        String customUrl = "";
+
+        // if "apiUrl" parameter was supplied, parse and normalize it for later use
+        if (!StringUtils.isEmpty(apiUri)) {
+            java.net.URI uri;
+
+            try {
+                uri = new URI(apiUri);
+            } catch (URISyntaxException ex) {
+                throw new ServiceException.BadRequestException(new ErrorMessage(400, "Invalid URI: " + apiUri));
+            }
+
+            customUrl = uri.toString();
```

Comment:
You don't really need it. It's not like `new URI()` does any normalization, I guess you are doing it for the reason to fail fast. You can avoid using `customUrl` variable and simply use apiUrl variable for all purposes.